Win32-network/test/Test/Async/Socket.hs

```diff
@@ -94,6 +96,27 @@ test_interruptible_accept =
                     (Just ThreadKilled)
                     (fromException e :: Maybe AsyncException)
 
+
+-- | Verify that closing a socket will terminate the `Network.Socket.accept`
+-- call.
+--
+test_close_accept :: IO ()
+test_close_accept =
+    bracket
+      (Socket.socket Socket.AF_INET Socket.Stream Socket.defaultProtocol)
+      Socket.close
+      $ \fd -> do
+        v <- newEmptyMVar
+        let addr = SockAddrInet 0 (Socket.tupleToHostAddress (127, 0, 0, 1))
+        Socket.bind fd addr
+        Socket.listen fd 1
+        _ <- forkIO $
+          void (Socket.accept fd)
+          `finally` putMVar v ()
+        Socket.close fd
```

Comment:
You need a thread delay first here (or another MVar sync dance), otherwise the socket may be closed first and then its not testing anything.

Later: oh I see this is what the next patch does, ok.

Reply:
Also a `yield` should be enough, isn't it?